runtime(termdebug): add no_tty option to prevent creation of TTY window

d-k-c opened the pull request:
In some circumstances, like debugging an embedded system, the TTY window is useless and there was no way to prevent its creation.

Add a new "no_tty" option to termdebug_config to let user get rid of TTY window if they don't want it.

This is based on development by @GitMensch in #10389, adapted to newer Vim Script.

Unfortunately I could not get the vim9 version to work with my local installation for unknown reasons, so it's untested.

yegappan commented on runtime/pack/dist/opt/termdebug/plugin/termdebug.vim:
```diff
@@ -354,23 +353,32 @@ enddef
-  if ptybufnr == 0
-    Echoerr('Failed to open the program terminal window')
-    return
+  var create_pty = 1
```
Instead of using 0 and 1, can you use the Boolean values true and false?

yegappan commented on runtime/pack/dist/opt/termdebug/plugin/termdebug.vim:
```diff
@@ -328,14 +328,13 @@ enddef
-  endif
+  var bufnames = ['debugged\ program', 'gdb\ communication', asmbufname, varbufname]
+  for bufname in bufnames
+    if bufnr(bufname) > 0 && bufexists(bufnr(bufname))
```
Instead of calling bufnr() twice, can you store the return value in a local variable and then use it?

yegappan commented on runtime/pack/dist/opt/termdebug/plugin/termdebug.vim:
```diff
@@ -587,37 +597,45 @@ def StartDebug_prompt(dict: dict<any>)
-      Echoerr('Failed to open the program terminal window')
-      job_stop(gdbjob)
-      return
+  var create_pty = 1
```
Can you use `var create_pty: bool = true`?

d-k-c committed:
runtime(termdebug): add no_tty option to prevent creation of TTY window

In some circumstances, like debugging an embedded system, the TTY window
is useless and there was no way to prevent its creation.

Add a new "no_tty" option to termdebug_config to let user get rid of TTY
window if they don't want it.

This is based on development by @GitMensch in #10389, adapted to newer
Vim Script.

Signed-off-by: Damien Riegel <damien@riegel.io>

d-k-c committed:
runtime(termdebug): close all buffers in the same way

For ASM and Variables buffer, check were done to make sure they existed
before attempting to close them, but not for debugged program or gdb
communication. The debugged program window is a user-facing one and
user might close it manually, so it's better to check if it exists.

Signed-off-by: Damien Riegel <damien@riegel.io>

d-k-c committed:
runtime(termdebug): restore TermDebugSendCommand being global

TermDebugSendCommand lost it's global visibility when converted to
vim9script. Restore it.

Fixes: 23f29ffc6427 ("runtime(termdebug): convert termdebug plugin to Vim9 script")

Signed-off-by: Damien Riegel <damien@riegel.io>

Shane-XB-Qian commented on runtime/pack/dist/opt/termdebug/plugin/termdebug.vim:
```diff
@@ -717,7 +717,7 @@ def ContinueCommand()
 
 # This is global so that a user can create their mappings with this.
-def TermDebugSendCommand(cmd: string)
+def g:TermDebugSendCommand(cmd: string)
```
if it worked, then you may need to update the doc/help as well, there is some description about this func.

d-k-c replied:
From my script, I can `call TermDebugSendCommand("...")` without the leading `g:`, but if the function is not declared with the `g:` then the call fails with `E117: Unknown function: TermDebugSendCommand`. I'm not sure the doc needs to be updated.
